### PR TITLE
chore: add test cases for getTaskByQuestId service

### LINKED ISSUE #619: test: `/get_tasks?quest_id=${questId}&addr=${address}`
read description here - https://github.com/starknet-id/starknet.quest/issues/602#issue-2255907921

### PR DESCRIPTION
**Pull request by leevigh**
I changed the apiService.test.js file to test the getTasksByQuestId API service. The test included checks for:
- when API returns no response
- when API returns a response in an unexpected format
- case when endpoint values are undefined and null
- when response is successful

# Pull Request type
chore: add test case

Please add the labels corresponding to the type of changes your PR introduces: Testing.
- Testing

Resolves: #619

### CODE REVIEW

File: tests/services/apiService.test.js [hunk 1]
```diff
@@ -40,3 +40,87 @@ describe("fetchQuestCategoryData function", () => {
...
+  });
+
+  it("should handle undefined and null values", async () => {
+    const mockResponse = undefined
+
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockResponse),
+    })
+
+    let result = await getTasksByQuestId({ address: '2' })
```
Comment (ayushtom) [inline tests/services/apiService.test.js]: the result here are error strings which the api returns. Not undefined. I would suggest you to test these values in the api and then check if the mock returns you have added here are correct

File: tests/services/apiService.test.js [hunk 2]
```diff
@@ -40,3 +40,87 @@ describe("fetchQuestCategoryData function", () => {
...
+    expect(result).toBeUndefined()
+
+    result = await getTasksByQuestId({ questId: null, address: null })
+    
+    expect(result).toBeUndefined()
+  });
+
+  it("should fetch and return data for a valid task", async () => {
+    const mockData = [
+      {
```
Comment (ayushtom) [inline tests/services/apiService.test.js:121]: not a valid mock data response. Please confirm with api what you get here

**Comment by ayushtom** [comment]
can you please resolve conflicts?

**Comment by leevigh** [comment]
> can you please resolve conflicts?

I'm done resolving the conflicts

**Review by ayushtom — APPROVED** [review]
test cases lgtm

**Comment by ayushtom** [comment]
can you please upload the test command output so i can verify if this covers the api call you were assigned

**Review by ayushtom — CHANGES_REQUESTED** [review]
you need to also add the typings of the response format on the front like i mention in the issue

**Comment by leevigh** [comment]
> can you please upload the test command output so i can verify if this covers the api call you were assigned

My tests cover lines 239-255 of apiService.ts


Screenshot 2024-04-26 at 10 05 48

**Comment by leevigh** [comment]
> you need to also add the typings of the response format on the front like i mention in the issue

The types are included in the getTasksByQuestId service response data i.e UserTask[] | QueryError. 
Could you please clarify what typings you are referring to in case I have misunderstood it?

**Review by ayushtom — APPROVED** [review]
lgtm

**Pull request merged**